src/main/java/com/example/diplomnabe/Controllers/RecipeController.java

```diff
@@ -0,0 +1,56 @@
+package com.example.diplomnabe.Controllers;
+import com.example.diplomnabe.Classes.Recipe;
+import com.example.diplomnabe.Classes.User;
+import com.example.diplomnabe.DTO.RecipeDTO;
+import com.example.diplomnabe.DTO.UserDTO;
+import com.example.diplomnabe.Repositories.UserRepository;
+import com.example.diplomnabe.Services.RecipeService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping(path = "api/version1/recipe")
+public class RecipeController
+{
+    private final RecipeService recipeService;
+
+    @Autowired
+    public RecipeController(RecipeService recipeService)
+    {
+        this.recipeService = recipeService;
+    }
+
+    @GetMapping
+    public List<RecipeDTO> getRecipes()
+    {
+        //return recipeService.getRecipes();
+        List<RecipeDTO> list_of_DTO = new ArrayList<>();
+        List<Recipe> recipes = recipeService.getRecipes();
+        for (Recipe recipe : recipes) {
+            list_of_DTO.add(recipe.convertRecipeToRecipeDTO());
+        }
+        return list_of_DTO;
+    }
+
+    @PostMapping("/create/{userId}")
```

Comment:
Try not to use a post verbs in your routes and for post routes to use path paramaters.
Add them to the payload and validate it.